Windows: Stop ignoring all deprecated warnings

unknownbrackets opened the pull request:
This removes some globally disabled warnings.  Was helpful for #14176.

Some network related warnings (mainly inet_nota and the WSA version) are exposed.  Also two `GetVersion()` calls.  In theory, if we update dependencies (say, glslang, spirv, etc.) this will better expose explicitly deprecated usage.

-[Unknown]

unknownbrackets committed:
Windows: Stop ignoring all deprecated warnings.

hrydgard commented:
Yes, this is a better way, nice one.

UWP is unhappy - I think you'll need to do the same in those vcprojs. Unfortunately I wasn't able to find a way to get UWP builds from the same project files to work, which is indeed really silly.

unknownbrackets commented:
I did change UWP but didn't try to build it, must've missed a file.  Let's see if this works with UI_UWP changed too.  Also seemed to error on deprecated instead of warning...

-[Unknown]

hrydgard commented:
Oh, sorry, didn't look carefully enough at that diff. Yeah let's hope that was it...

unknownbrackets committed:
UWP: Disable SDLChecks, which error on deprecated.

We should get rid of deprecated usage, but it's troublesome for only UWP
to compile differently.  These were ignored before.
